### PR TITLE
Handle the case of external endpoints for the apollo linker

### PR DESCRIPTION
**Pull request by ahafidi**
We're now letting external GraphQL endpoint, but the user should then provide an extra argument to the Prismic link for knowing the repository name in the case where an external endpoint is provided. Example:
```
PrismicLink({
  uri: 'https://mycustomdomain.com/graphql',
  accessToken: 'my_access_token', // could be undefined
  repositoryName: 'my-prismic-repository'
})
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,55 +2,68 @@ import { HttpLink } from 'apollo-link-http';
 import { setContext } from 'apollo-link-context';
 import Prismic from 'prismic-javascript';
 
-const PRISMIC_ENDPOINT_REG = /^(https?):\/\/([^.]+)\.(cdn.)?([^.]+\.[^.]+)\/graphql\/?$/;
+const PRISMIC_ENDPOINT_REG = /^https?:\/\/([^.]+)\.(?:cdn.)?prismic.io\/graphql\/?/;
```
Comment (MarcMcIntosh) [inline index.js]: Support wroom.io addresses as well
``` 
^https?:\/\/([^.]+)\.(?:cdn.)?(wroom.(test|io)|prismic.io)\/graphql\/?/;
```

Comment (MarcMcIntosh) [reply index.js]: This will break the if statement below

Comment (MarcMcIntosh) [reply index.js]: the length will be 4

File: index.js [hunk 2]
```diff
@@ -2,55 +2,68 @@ import { HttpLink } from 'apollo-link-http';
...
 
-function parseURI(endpoint) {
-  const [, protocol, repository, cdn, domain] = endpoint.match(PRISMIC_ENDPOINT_REG);
+function parseURI(endpointGraphQL) {
+  const tokens = endpointGraphQL.match(PRISMIC_ENDPOINT_REG);
+
+  if (tokens !== null && Array.isArray(tokens) && tokens.length === 2) {
```
Comment (MarcMcIntosh) [inline index.js]: with length will be 4 if regexp is changed

File: index.js [hunk 3]
```diff
@@ -2,55 +2,68 @@ import { HttpLink } from 'apollo-link-http';
 import { setContext } from 'apollo-link-context';
 import Prismic from 'prismic-javascript';
 
-const PRISMIC_ENDPOINT_REG = /^(https?):\/\/([^.]+)\.(cdn.)?([^.]+\.[^.]+)\/graphql\/?$/;
+const PRISMIC_ENDPOINT_REG = /^https?:\/\/([^.]+)\.(?:cdn.)?prismic.io\/graphql\/?/;
 
-function parseURI(endpoint) {
-  const [, protocol, repository, cdn, domain] = endpoint.match(PRISMIC_ENDPOINT_REG);
+function parseURI(endpointGraphQL) {
```
Comment (srenault) [inline index.js]: I think we should have a `parsePrismicEndpoint` that returns:
```javascript
const prismicEndpoint = parsePrismicEndpoint(uri); // enforce cdn if prismic endpoint

if (!prismicEndpoint && !repositoryName) {
  throw new Error("Please provide repository name....");
}

const endpoint = prismicEndpoint || uri;
const apiEndpoint = `${endpoint}/api`;
const graphqlEndpoint = `${endpoint}/graphql`;
```

```bash
parsePrismicEndpoint(“https://jojo.prismic.io/graphql“) => “https://jojo.cdn.prismic.io/graphql”
parsePrismicEndpoint(“https://jojo.wroom.io/graphql“) => “https://jojo.cdn.wroom.io/graphql”
parsePrismicEndpoint(“https://jojo.wroom.test/graphql“) => “https://jojo.cdn.wroom.test/graphql”
parsePrismicEndpoint(“https://google.fr“) => null
```

**Comment by MarcMcIntosh** [comment]
Regexp is good :)

**Review by srenault — APPROVED** [review]

**Comment by MarcMcIntosh** [comment]
found 14731 high severity vulnerabilities
run `npm audit fix` to fix them, or `npm audit` for details

**Comment by MarcMcIntosh** [comment]
All good :)

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**